go/vt/tabletmanager/rpc_replication.go

```diff
@@ -555,14 +547,63 @@ func isMasterEligible(tabletType topodatapb.TabletType) bool {
 	return false
 }
 
-func (agent *ActionAgent) enableSemiSync(master bool) error {
+func (agent *ActionAgent) setSemiSync(tabletType topodatapb.TabletType) error {
```

Comment:
I'd recommend a different method name. "setSemiSync" suggests setting it to while you may actually reset it. Maybe "adjustSemiSync", or "fixSemiSync"?